File: Front/ProEventos-App/src/app/models/ItemCotacao.ts
import { Cotacao } from './Cotacao';
import { ItemPedido } from './ItemPedido';
import { SolicitacaoProduto } from './SolicitacaoProduto';

export class ItemCotacao {
   id: number;
   idCotacao: number;
   idSolicitacaoProduto: number;
   solicitacaoProduto: SolicitacaoProduto;
   idProduto: number;
   qtdeProduto: number;
   precoUnit: number ;
   totalItem: number ;
   cotacaoId: number;
   cotacao: Cotacao;
   itemPedidoId: number;
   itemPedido: ItemPedido;

}

File: Back/src/SistemaCompra.Domain/ItemCotacao.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SistemaCompra.Domain
{
    public class ItemCotacao
    {
        public int Id { get; set; }
        public int IdCotacao { get; set; }
        public int IdSolicitacaoProduto { get; set; }
        public SolicitacaoProduto SolicitacaoProduto { get; set; }
        public int IdProduto { get; set; }
        public int QtdeProduto { get; set; }
        public double PrecoUnit { get; set; }
         public double TotalItem { get; set; }
        public int cotacaoId { get; set; }
        public Cotacao Cotacao { get; set; }
         [ForeignKey("ItemPedido")]public int itemPedidoId { get; set; }
        public ItemPedido itemPedido { get; set; }

    }
}